openquakeplatform_taxtweb/utils/taxtweb_eng.py

```diff
@@ -0,0 +1,4037 @@
+    return n + (" is" if n <= 1 else " are") + " given."
+
+class TaxtSel(object):
+    def __init__(self, name, items=[], val=-1, disabled=False, change_cb=None):
+        '''
+        items => list of 1 element dicts
+        val => int identifing current val item
+        '''
+        self._name = name
+        self._items = items[:]
+        if items:
+            if val == -1:
+                self._val = 0
+            else:
+                self._val = val
+        else:
+            self._val = val
+
+        self._disabled = disabled
+        self._first_disabled = False
+        self._change_cb = change_cb
+
+    def empty(self):
+        self._items = []
+        self._val = -1
+
+    def disabled(self, disabled=None):
+        if disabled is None:
+            return self._disabled
+        else:
+            self._disabled = disabled
+
+    def first_disabled(self, first_disabled=None):
+        if first_disabled is not None:
+            self._first_disabled = first_disabled
+
+        return self._first_disabled
+
+    def items(self, items=None, val=0):
+        '''
+        items => list of 1 element dicts
+        val => int identifing current val item
+        '''
+        if items == None:
+            return self._items
+        else:
+            self._items = items[:]
+
+        self.val(val)
+
+    def val(self, val=-1):
+        if val != -1:
+            if isinstance(val, str):
+                val = self.items.index(val)
+
+            if val < -1 or val > len(self._items):
+                raise ValueError
+
+            self._val = val
+            if self._change_cb:
+                self._change_cb(self)
+
+        return self._val
+
+    def __str__(self):
+        ret = "  %s (Sel)\n" % self._name
+        if self._items:
+            for k, v in enumerate(self._items):
+                if k == self._val:
+                    ret += "    %02d) * [%s]\n" % (k, v)
+        else:
+            ret += "    - no elements -\n"
+
+        return ret
+
+
+class TaxtBool(object):
+    def __init__(self, name, val=False, change_cb=None):
+        self._name = name
+        self._val = val
+        self._change_cb = change_cb
+
+    def val(self, val=None):
+        if val == None:
+            return self._val
+        else:
+            self._val = val
+            if self._change_cb:
+                self._change_cb(self)
+            return self._val
+
+    def checked(self, val=None):
+        return self.val(val)
+
+    def __str__(self):
+        return "  %s (Bool)\n    %s\n" % (self._name, "True" if self._val else "False")
+
+
+class TaxtRadioItem(object):
+    def __init__(self, val=None, checked=False, radio=None, change_cb=None):
+        self._val = val
+        self._checked = checked
+        self._change_cb = change_cb
+        self._radio = radio
+
+    def radio(self, radio=None):
+        if (radio == None):
+            return self._radio
+        else:
+            self._radio = radio
+
```

Comment:
`val == None` and `val != None` is considered bad practice. The Pythonic way is `val is None` or `val is not None`.